projects/packages/classic-theme-helper/src/custom-post-types/class-nova-restaurant.php

```diff
@@ -0,0 +1,1758 @@
+			// If we want paging in the future, we'll need to add the taxonomy sort here (or at least before the DB query is made)
+			$query->query_vars['posts_per_page'] = -1;
+		}
+
+		/**
+		 * Custom sort the menu item queries by menu taxonomies.
+		 *
+		 * @param WP_Post[] $posts Array of post objects.
+		 * @param WP_Query  $query The WP_Query instance.
+		 *
+		 * @return WP_Post[]
+		 */
+		public function sort_menu_item_queries_by_menu_taxonomy( $posts, $query ) {
+			if ( ! $posts ) {
+				return $posts;
+			}
+
+			if ( ! $this->is_menu_item_query( $query ) ) {
+				return $posts;
+			}
+
+			$grouped_by_term = array();
+
+			foreach ( $posts as $post ) {
+				$term = $this->get_menu_item_menu_leaf( $post->ID );
+				if ( ! $term || is_wp_error( $term ) ) {
+					$term_id = 0;
+				} else {
+					$term_id = $term->term_id;
+				}
+
+				if ( ! isset( $grouped_by_term[ "$term_id" ] ) ) {
+					$grouped_by_term[ "$term_id" ] = array();
+				}
+
+				$grouped_by_term[ "$term_id" ][] = $post;
+			}
+
+			$term_order = get_option( 'nova_menu_order', array() );
+
+			$return = array();
+			foreach ( $term_order as $term_id ) {
+				if ( isset( $grouped_by_term[ "$term_id" ] ) ) {
+					$return = array_merge( $return, $grouped_by_term[ "$term_id" ] );
+					unset( $grouped_by_term[ "$term_id" ] );
+				}
+			}
+
+			foreach ( $grouped_by_term as $term_id => $posts ) {
+				$return = array_merge( $return, $posts );
+			}
+
+			return $return;
+		}
+
+		/**
+		 * Add new "Add many items" submenu, custom colunmns, and custom bulk actions.
+		 *
+		 * @return void
+		 */
+		public function add_admin_menus() {
+			$hook = add_submenu_page(
+				'edit.php?post_type=' . self::MENU_ITEM_POST_TYPE,
+				__( 'Add Many Items', 'jetpack-classic-theme-helper' ),
+				__( 'Add Many Items', 'jetpack-classic-theme-helper' ),
+				'edit_pages',
+				'add_many_nova_items',
+				array( $this, 'add_many_new_items_page' )
+			);
+
+			add_action( "load-$hook", array( $this, 'add_many_new_items_page_load' ) );
+
+			add_action( 'current_screen', array( $this, 'current_screen_load' ) );
+
+			/*
+			* Adjust 'Add Many Items' submenu position
+			* We're making changes to the menu global, but no other choice unfortunately.
+			* phpcs:disable WordPress.WP.GlobalVariablesOverride.Prohibited
+			*/
+			if ( isset( $GLOBALS['submenu'][ 'edit.php?post_type=' . self::MENU_ITEM_POST_TYPE ] ) ) {
+				$submenu_item = array_pop( $GLOBALS['submenu'][ 'edit.php?post_type=' . self::MENU_ITEM_POST_TYPE ] );
+				$GLOBALS['submenu'][ 'edit.php?post_type=' . self::MENU_ITEM_POST_TYPE ][11] = $submenu_item;
+				ksort( $GLOBALS['submenu'][ 'edit.php?post_type=' . self::MENU_ITEM_POST_TYPE ] );
+			}
+			// phpcs:enable WordPress.WP.GlobalVariablesOverride.Prohibited
+
+			$this->setup_menu_item_columns();
+
+			Assets::register_script(
+				'nova-menu-checkboxes',
+				'../../dist/custom-post-types/js/menu-checkboxes.js',
+				__FILE__,
+				array(
+					'dependencies' => array(
```

Comment:
Is jQuery needed here? I don't see it nova.php in trunk https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/custom-post-types/nova.php#L587

Reply:
Good catch, I just missed removing it here. Fixed in bd443351cfe77a3b2186e44c8f6139c64157ca21